streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1269,20 +1280,36 @@ void handleTopologyUpdates() {
         }
     }
 
+    static class RecordsProcessedMetadata {
```

Comment:
See my other comment: I think we can avoid propagating both processed-records and processed-bytes from the `process` call.